rocketpy/simulation/monte_carlo.py

```diff
@@ -1764,28 +1319,36 @@ def reprint(self, sim_idx, end="\n", flush=False):
         ----------
         msg : str
             Message to be printed.
-        end : str, optional
-            String appended after the message. Default is a new line.
-        flush : bool, optional
-            If True, the output is flushed. Default is False.
 
         Returns
         -------
         None
         """
-        average_time = (time() - self.initial_time) / (self.count - self.initial_count)
-        estimated_time = int(
-            (self.n_simulations - (self.count - self.initial_count)) * average_time
-        )
+        average_time = (time() - self.start_time) / (self.count - self.initial_count)
+        estimated_time = int((self.n_simulations - self.count) * average_time)
 
         msg = f"Current iteration: {sim_idx:06d}"
         msg += f" | Average Time per Iteration: {average_time:.3f} s"
         msg += f" | Estimated time left: {estimated_time} s"
 
-        len_msg = len(msg)
-        if len_msg < self._last_print_len:
-            msg += " " * (self._last_print_len - len_msg)
-        else:
-            self._last_print_len = len_msg
+        MonteCarlo._reprint(msg, end="\r", flush=True)
+
+    def finalize_status(self):
+        """Prints the final status of the simulation.
+
+        Parameters
+        ----------
+        show_process_time : bool, optional
+            If True, the process time will be shown. Default is True.
+
+        Returns
+        -------
+        None
+        """
```

Comment:
The docstring is outdated

```suggestion
        """Prints the final status of the simulation."""
```

I also suggest renaming the method. "print_final_status" seems cooler.

Reply:
Solved in [fe7bad3](https://github.com/RocketPy-Team/RocketPy/pull/649/commits/fe7bad3f0ec74927083290a3436b8ac5aaf0afef).